### ISSUE TITLE
Hangs on Oculus Quest2 v51

### ISSUE THREAD

**Issue by eq-soft** [DrBeef/Lambda1VR#96]
The game hangs on level loading (savegame loading, new game start - doesn't matter) right after updating Oculus Quest 2 firmware to v51

**Comment by baggyg** [comment]
Was it working before you upgraded? One of the team has v51 and has tested Lambda1VR and confirmed it is still working. Try a headset restart and let us know. 

Have you moved the full game files across to the correct place on the headset?

**Comment by eq-soft** [comment]
Yes, it was working fine before update. I just finished playing, turned oculus to standby mode and after few days i got update and the same game starts to hang. Probably it's something broken on my side, but i changed nothing before.

I already heard that it works fine, but i clarified it means "the game launches", but not "you can load level and play".

Probably i can check some logs on my device?

**Comment by baggyg** [comment]
Yes, we've confirmed it loading a level, not just starting up. In the first instance I would uninstall and install again via Sidequest. Don't delete the files, just uninstall the app. 

I know its common sense but do make sure you have restarted the headset at least once. Once an app crashes it hangs in the background and every time you start you are just launching the frozen version. The headset needs to properly power down (not just to standby), then back on.

**Comment by eq-soft** [comment]
Yes, i already tried to reinstall apk few times - i thought my apk is outdated and to give permission to access file system again. And yes, i restarted my oculus because hanged app breaks the oculus menu too, so i had to restart the device to try again :)
Ok, it means that it's really not a common issue, but it's strange that i really nothing changed. I will try to find some logs and reupload data folder to the file system.

**Comment by baggyg** [comment]
v51 is only just rolling out so we'll collect some more feedback. Recopying data files is a good idea as well. We'll keep this issue open until we get to the bottom of it.

**Comment by eq-soft** [comment]
I will update it after recopying. Thank you!

**Comment by eq-soft** [comment]
Unfortunately no luck. What i did:
1. Renamed my old xash folder to xash_2 (i have uncompleted saves :))
2. Removed lambda1vr app in oculus apps menu
3. Installed last version apk from this repo
4. Downloaded xash archive from big known site
5. Copied xash directory to /sdcard/xash/ and ensured that i have /sdcard/xash/valve structure
6. Launched lambda1vr: it requests filesystem permission, i grant it and checked "Remember" option. The game shows menu, i click on "New game" -> "Medium" and it immediately freezes right in the menu. I waited for few minutes (hoped to long loading).
---
1. Removed xash directory, renamed my old xash_2 to xash (with valve folder inside)
2. Launched lambda1vr: it shows menu, i click on "New game" -> "Meduim", it show me "Loading" screen and then freezes. I also waited for few minutes.

So sad, i must finish the game :)

**Comment by baggyg** [comment]
Can you confirm if you are on the PTC branch of OS releases? We've received reports there sometimes there are issues with those (since they are betas), but not actual releases. 

In terms of your progress, I don't understand step 4 at all. You definitely should not create a xash folder yourself. I understand renaming the current one as a backup, but just start the app and it will create what it needs (and vitally, nothing else). 

Check the PTC setting for use please (via the mobile app) and make sure it is turned off

**Comment by eq-soft** [comment]
PTC option is on. I think turning off will change nothing now, because some beta version is already installed?
Now my firmware is 53.0.0.98.132.466624179
And the system version is 50500320039500150

About step 4: all game archives (with "valve" content) already contain xash folder so it's much easier to just copy the whole xash instead of launching the client to create the folder and then copy valve folder inside. Or maybe the game creates the folder with some special attributes?

**Comment by eq-soft** [comment]
Just turned it off but nothing happens. I also restarted device - nothing. Will wait, probably it will rollback in standby mode.

**Comment by eq-soft** [comment]
And now i finally got v51 (device just made factory reset itself, it was big surprise for me).
I installed Lambda1vr 1.5.1, copied xash directory to the /sdcard, runned it, and it hangs again (just in menu, without "Loading" screen).
Also i saw reports from users that some other games, worked on v51, don't work on their devices with v51. Probably there is some unknown condition for success.

**Comment by eq-soft** [comment]
I was wrong: i launched all my favorite games on v51, but not lambda1vr. It still hangs.

**Comment by baggyg** [comment]
Again, I'm not sure why you are copying the Xash folder. However there is an update for Lambda1VR dropping very soon, so I would recommend checking that out (next couple of days). 

We're not aware of any of our ports having issues on v51

**Comment by eq-soft** [comment]
> Again, I'm not sure why you are copying the Xash folder.

I'm doing it because of Installation instructions on the lambda1vr.com:
* Create a folder on the root of your Quest's internal memory called "xash" (You will need to connect your Quest via a USB cable and enable MTP to do this).
* Locate the install of half-life on your PC, if installed from Steam then it will be somewhere like C:/Program Files (x86)/Steam/steamApps/common/HalfLife/
* Copy the entire folder "valve" to the newly created "xash" folder on the Quest (This will take a long time). I recommend doing this via standard Windows Explorer rather than SideQuest.

**Comment by baggyg** [comment]
Ok, thats on us, We need to update that.... someone else has confirmed that creating the folder yourself causes issues. Probably something stricter on v51 around file names (possible case sensitivity). 

Take a backup and delete the entire xash folder. 

Then run the app once and wait until you see the menu (took about 10 seconds for me)

Exit. 

You should then have the xash and valve folder. Copy the assets over and then confirm working. Once working, then copy your saves from the backup.

**Comment by baggyg** [comment]
I've just installed from scratch on v51 and didn't have any issues. The loading circle takes a good 5-7 seconds and for some reason the menu appears to my right whenever I open but otherwise fine. 

Try the new version dropping soon and let us know if its still an issue

**Comment by baggyg** [comment]
New update is out to try

**Comment by eq-soft** [comment]
Still no luck. What i did:
1. Renamed `xash` folder to `xash2`
2. Launched `lambda1vr` app - got menu with black background
3. Checked sdcard directory: `xash` folder was successfully created by app. By the way i compared `xash2` and `xash` attributtes - they are identical
4. Moved `valve` folder from `xash2` to `xash`
5. Launched `lambda1vr` app - got menu with game background (it means the game has access to the folder)
6. Clicked New game > Meduim - and it freeses
8. Uninstalled `1.5.1`, Installed `1.6.2`
9. Launched `lambda1vr` app - now startup is a little longer on three dots stage and now i see flat menu window instead of beneded as before
10. Clicked New game > Medium - it freezes.

Also i found `engine.log`:
```
================================================================================
	Xash3D FWGS (build 2952, Android-aarch64) started at May01 2023 [19:39.29]
================================================================================
[2023:05:01|19:39:29] Cannot load xashcomm.lst
[2023:05:01|19:39:30] Video: GL4ES wrapper
[2023:05:01|19:39:30] ^3Warning:^7 GL_UploadTexture: cubemaps isn't supported, *vsdct ignored
[2023:05:01|19:39:31] Localize_AddToDict: loaded 241 words from resource/gameui_english.txt
[2023:05:01|19:39:31] Localize_AddToDict: loaded 190 words from resource/valve_english.txt
[2023:05:01|19:39:31] Reading config script file settings.scr
[2023:05:01|19:39:31] Reading config script file user.scr
[2023:05:01|19:39:31] ^3Warning:^7 FS_LoadImage: couldn't load "sprites/scope"
[2023:05:01|19:39:31] AVI: Not supported
[2023:05:01|19:39:31] vgui_support: not found
[2023:05:01|19:39:32] Loading new-style background from game succeded
[2023:05:01|19:39:32] Uploaded RobotoCondensed_47_1000_font.bmp to 147
[2023:05:01|19:39:32] Rendering RobotoCondensed(47, 1000) took 0.221059 seconds
[2023:05:01|19:39:32] Uploaded RobotoCondensed_36_1000_font.bmp to 148
[2023:05:01|19:39:32] Rendering RobotoCondensed(36, 1000) took 0.203274 seconds
[2023:05:01|19:39:33] Uploaded RobotoCondensed_73_1000_font.bmp to 149
[2023:05:01|19:39:33] Rendering RobotoCondensed(73, 1000) took 0.221886 seconds
[2023:05:01|19:39:33] Uploaded RobotoCondensed_47_1000_font.bmp to 147
[2023:05:01|19:39:33] Rendering RobotoCondensed(47, 1000) took 0.005218 seconds
[2023:05:01|19:39:33] Uploaded DroidSans_32_500_o1_font.bmp to 150
[2023:05:01|19:39:33] Rendering DroidSans(32, 500) took 0.232983 seconds
[2023:05:01|19:39:33] Reading config script file settings.scr
[2023:05:01|19:39:33] Reading config script file user.scr
[2023:05:01|19:39:33] execing config.cfg
[2023:05:01|19:39:33] execing keyboard.cfg
[2023:05:01|19:39:33] execing valve.rc
[2023:05:01|19:39:33] execing language.cfg
[2023:05:01|19:39:33] execing autoexec.cfg
[2023:05:01|19:39:33] execing violence.cfg
[2023:05:01|19:39:33] execing config.cfg
[2023:05:01|19:39:33] execing keyboard.cfg
[2023:05:01|19:39:34] ^3Warning:^7 FS_LoadImage: couldn't load "sprites/scope"
[2023:05:01|19:39:38] Dll loaded for mod Half-Life
[2023:05:01|19:39:38] Server loaded
[2023:05:01|19:39:38] Spawn Server: c0a0 
```

I think there is some extremely stupud mistake from my side, but i have no idea, where it is :)

**Comment by eq-soft** [comment]
Confirmed: level loading is really long:
```
[2023:05:01|20:03:37] Spawn Server: c0a0
[2023:05:01|20:04:57] execing skill.cfg
[2023:05:01|20:04:57] server: GAME SKILL LEVEL:2
[2023:05:01|20:04:57] ^1Error:^7 Mod_ForName: models/barney_vest.mdl couldn't load
[2023:05:01|20:04:58] ^1Error:^7 Mod_ForName: models/barney_helmet.mdl couldn't load
[2023:05:01|20:05:00] server: *Graph Loaded!
[2023:05:01|20:05:00] SV_LoadFromFile: 0 entities inhibited
[2023:05:01|20:05:00] server: Dead end link trainstop27
[2023:05:01|20:05:00] Game started
[2023:05:01|20:05:00] Spooling demo header.
[2023:05:01|20:05:01] ^1Error:^7 Mod_ForName: models/barney_vest.mdl couldn't load
[2023:05:01|20:05:01] ^1Error:^7 Mod_ForName: models/barney_helmet.mdl couldn't load
[2023:05:01|20:05:02] server: **Graph Pointers Set!
[2023:05:01|20:05:02] Level loaded in 84.58 sec
```

```
[2023:05:01|20:06:09] server: CHANGE LEVEL: c0a0a c0a0toa
[2023:05:01|20:06:10] Spawn Server: c0a0a [c0a0toa]
[2023:05:01|20:07:14] Loading game from save/c0a0a.hl1...
[2023:05:01|20:07:14] ERROR: couldn't open.
[2023:05:01|20:07:14] server: GAME SKILL LEVEL:2
[2023:05:01|20:07:15] ^1Error:^7 Mod_ForName: models/barney_vest.mdl couldn't load
[2023:05:01|20:07:15] ^1Error:^7 Mod_ForName: models/barney_helmet.mdl couldn't load
[2023:05:01|20:07:15] SV_LoadFromFile: 0 entities inhibited
[2023:05:01|20:07:15] Loading game from save/c0a0.hl1...
[2023:05:01|20:07:15] Transferring player (1)
[2023:05:01|20:07:15] Merging changes for global: func_tracktrain
[2023:05:01|20:07:15] Loading decals from c0a0
[2023:05:01|20:07:15] Loading game from save/c0a0b.hl1...
[2023:05:01|20:07:15] ERROR: couldn't open.
[2023:05:01|20:07:15] Game started
[2023:05:01|20:07:15] ^6Server restarted
[2023:05:01|20:07:15] Spooling demo header.
[2023:05:01|20:07:16] ^1Error:^7 Mod_ForName: models/barney_vest.mdl couldn't load
[2023:05:01|20:07:16] ^1Error:^7 Mod_ForName: models/barney_helmet.mdl couldn't load
[2023:05:01|20:07:16] Level loaded in 66.56 sec
```

As far as i remember it was much faster before, or i'm wrong?

**Comment by baggyg** [comment]
We've seen some spotty reports of v51 / v53 acting slower, but then also reports that this goes away after a while. Unfortunately for us this is entirely on the Oculus OS side and things they are changing and nothing to do with the ports. 

Are you using the HD textures?

**Comment by eq-soft** [comment]
As far as i know, HD textures is `Allow materials` option? I tested it and got some results:
Starting new game with `Allow materials` option - 80-100 sec;
Starting new game without `Allow materials` option - 35-45 sec;
Loading saved game and reloading this saved game again - 1.5 sec.
Changing the level takes the same time as loading first level after application start.

Unfortunately it still not gone away in my case :)

**Comment by baggyg** [comment]
None of us are seeing times anywhere near that, either on v51 or v53. 

Xash is very small file orientated, but shouldn't be that long.

**Comment by baggyg** [comment]
We have had reports of loading time issues. This is due to OS changes by Meta / Oculus especially on the Public test Channel (PTC) and nothing to do with the apps themselves (people have also reported this in other store apps). In cases where it has appeared, it has also seemed to disappear either with the next update or somewhat randomly. 

Unfortunately this is completely out of our control and its down to Oculus to fix the underlying bug, which they may well have done in more recent versions. If you have this issue, we suggest making sure you are on the latest OS available to you, restarting the headset and trying again. Some users has found that uninstalling and reinstalling the app has helped (no need to delete the files themselves, so save games will be retained).

**Comment by eq-soft** [comment]
@baggyg, now i really fixed it.

First of all I noticed some other games bugs, so i did factory reset. Unfortunately it did not solve the long loadings.
I found that the valve folder was ~2.5Gb (localized version with improved textures). I got original distro that was ~450Mb (original HD version). I uploaded it and finally it started to take for about 3 seconds to load any level.

So it seems that after firmware update original versions continued to work fine (i think they also loaded longer than before, but 3 seconds is still good time to be ignored), but loading of seriously improved versions (at least custom HD textures with own sounds) become dramatically slow.

This can explain why "some people reported" this issue.
